enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/context.tsx

```diff
@@ -0,0 +1,40 @@
+import {
+  createContext,
+  type PropsWithChildren,
+  useContext,
+  useMemo,
+} from "react";
+
+import type { SdkPluginsConfig } from "embedding-sdk/lib/plugins";
+import { initializePlugins } from "embedding-sdk/lib/plugins/initialize";
+import type Question from "metabase-lib/v1/Question";
+
+type InteractiveDashboardContextType = Partial<{
+  plugins: SdkPluginsConfig;
+  onEditQuestion?: (question: Question) => void;
+}>;
+const InteractiveDashboardContext =
+  createContext<InteractiveDashboardContextType>({});
+
+export const InteractiveDashboardProvider = ({
+  children,
+  plugins,
+  onEditQuestion,
+}: PropsWithChildren<InteractiveDashboardContextType>) => {
+  const value = useMemo(
+    () => ({
+      plugins: initializePlugins(plugins),
+      onEditQuestion,
+    }),
```

Comment:
This is only part of the solution. Since `plugins` and `onEditQuestion` aren't memoized, this part won't have any effect. I think you can either.
1. Memoize `plugins` and `onEditQuestion` before passing them in or
2. pass some of the values individually and have the provider memoize them itself, e.g. instead of passing `plugins` you pass both `plugins` and `globalPlugsin` (or better yet, you can call `useSdkSelector(getPlugins)` in the provider 🤷 , but for `onEditQuestion` it still has to be memoized before passing down to this provider.